controllers/clusters/cassandra_controller_test.go

```diff
@@ -106,4 +109,53 @@ var _ = Describe("Cassandra Controller", func() {
 			}, timeout, interval).Should(BeTrue())
 		})
 	})
+
+	When("Deleting the Cassandra resource by avoiding operator", func() {
+		It("should try to get the cluster details and receive StatusNotFound", func() {
+			cassandraManifest2 := cassandraManifest.DeepCopy()
+			cassandraManifest2.Name += "-test-external-delete"
+			cassandraManifest2.ResourceVersion = ""
+
+			cassandra2 := v1beta1.Cassandra{}
+			cassandra2NamespacedName := types.NamespacedName{
+				Namespace: cassandraManifest2.Namespace,
+				Name:      cassandraManifest2.Name,
+			}
+
+			Expect(k8sClient.Create(ctx, cassandraManifest2)).Should(Succeed())
+
+			doneCh := tests.NewChannelWithTimeout(timeout)
+
+			Eventually(func() bool {
+				if err := k8sClient.Get(ctx, cassandra2NamespacedName, &cassandra2); err != nil {
+					return false
+				}
+
+				if cassandra2.Status.State != models.RunningStatus {
+					return false
+				}
+
+				doneCh <- struct{}{}
+
+				return true
+			}, timeout, interval).Should(BeTrue())
+
+			<-doneCh
+
+			By("testing by deleting the cluster by Instaclustr API client")
```

Comment:
By("using all possible ways other than k8s to delete a resource, the k8s operator scheduler should notice the changes and set the status of the deleted resource accordingly.")

Reply:
Fixed